### PR TITLE
Allow setting paths through env vars

### PR DESCRIPTION
**Pull request by g-arjones**
This makes integration with some build environments easier (i.e by sourcing an env script)

### CODE REVIEW

File: cmake/FindFreeRTOS.cmake [hunk 1]
```diff
@@ -7,6 +7,10 @@ list(REMOVE_DUPLICATES FreeRTOS_FIND_COMPONENTS)
 
 set(FreeRTOS_HEAPS 1 2 3 4 5)
 
+if(NOT FREERTOS_PATH)
+    set(FREERTOS_PATH $ENV{FREERTOS_PATH} CACHE PATH "Path to FreeRTOS")
+endif()
+
```
Comment (atsju) [inline cmake/FindFreeRTOS.cmake:13]: This one coding scheme is inconsistant with the other 3.
With quick glance I do not expect any difference in behavior and I find this one clearer (personally).
What do you think to change the other 3 implementations to fit syntax of this one ?

**Review by atsju — COMMENTED** [review]
I like the idea 👍

File: cmake/FindFreeRTOS.cmake (hunk 1 above)
Comment (g-arjones) [reply cmake/FindFreeRTOS.cmake:13]: Sure.

**Comment by g-arjones** [comment]
Updated.

**Comment by atsju** [comment]
Hello @g-arjones I really like the idea of this PR and would like to merge it soon :) 
Could you please check if it makes to update the readme so that new users can find and use the feature without digging to much ?

**Comment by g-arjones** [comment]
@atsju I think it does, yes. Unfortunately, since the merge took too long (I'm using this in production) I couldn't wait and have been using a fork for a while now. I'm a bit overwhelmed and can't add documentation. Sorry.

I have enabled edits by maintainers so you are welcome to push docs to this branch if you are still interested in my contribution.

**Comment by atsju** [comment]
Thank you for fast and honest response. Maintenance should go a bit faster from now. Maybe you will be able to go back to this repo in a while. I will see how we manage this.

**Comment by Hish15** [comment]
Closing in favor of #226

**Pull request closed without merging**